refactor: rewrite of geometryutil as typescript

Dexus opened the pull request:
This refactoring needs #131 merged before

Dexus committed:
refactor: rewrite of geometryutil as typescript

Dexus committed:
refactor: remove geometryutil.js script references and update imports

Dexus committed:
refactor: remove geometryutil.js script

Dexus committed:
refactor: implement UMD module for geometryutil and update build process

Dexus committed:
refactor: add rollup-plugin-swc-minify for improved minification in build process

willmurnane commented on main/util/geometryutil.ts:
```diff
@@ -0,0 +1,1809 @@
+  return _internalLineIntersect(A, B, E, F, infinite);
+}
+
+export const QuadraticBezier = {
```
Should this be a `class`?

willmurnane commented on main/util/geometryutil.ts:
```diff
@@ -0,0 +1,1809 @@
+      t_c1 = c1,
+      t_c2 = c2,
+      t_p2 = p2;
+    const p1c1: Point = new Point(
+      t_p1.x + (t_c1.x - t_p1.x) * t,
+      t_p1.y + (t_c1.y - t_p1.y) * t,
+    );
```
Should define and use a `lerp` function that encapsulates the "partway from one point to another" behavior.

willmurnane commented on main/util/geometryutil.ts:
```diff
@@ -0,0 +1,1809 @@
+  },
+};
+
+export function getPolygonBounds(polygon: Polygon): Bounds | null {
```
This should be a member function of the `Polygon` class.

willmurnane reviewed (commented):
Most of these comments can be resolved later, but just commenting while I'm looking at this.

Dexus commented on helper_scripts/build_ts_and_umd.js:
```diff
@@ -0,0 +1,12 @@
+
+// Build the UMD modules using Rollup
+console.log('Building UMD modules with Rollup...');
+execSync('npx rollup -c rollup.config.mjs', { stdio: 'inherit' });
```
see https://github.com/deepnest-next/deepnest/pull/132/files#diff-73f5a0c098eeed918ec330da8ffc45e7f4ca600d9bb969d0a9dea4b5d2909c85R197

Dexus committed:
redefine bounds instead of single values

Dexus committed:
tests for geometryutil and refactor/cleanup the code

- Added unit tests for all functions in geometryutil.ts
- Refactored code for better readability and maintainability
- Removed unused imports and variables
- Improved type definitions and interfaces

polygon needs to updated where it called getPolygonBounds

Dexus commented:
ATTENTION: getPolygonBounds calls against GeometryUtil needs to updated the polygon calls for x,y, width and height
